src/router.js

```diff
@@ -26,4 +26,201 @@ router.post('/applications', async (request, response) => {
+    return undefined;
+  }
+
+  return month;
+};
+
+/**
+ * Clean a user supplied 'year' in to either a `number` or `undefined`.
+ *
+ * @param {string | undefined} year The user's supplied year value.
+ * @returns {number | undefined} The cleaned year value.
+ */
+const cleanYear = (year) => {
+  if (year === undefined) {
+    return undefined;
+  }
+
+  if (year < 1000) {
+    return undefined;
+  }
+
+  if (year > 9999) {
+    return undefined;
+  }
+
+  return year;
+};
+
+/**
+ * Clean the incoming POST request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {any} body the incoming request's body
+ * @returns {any} a json object that's just got our cleaned up fields on it
+ */
+const cleanInput = (body) => {
+  const firearmDate = new Date(body.certificateIssuedDate);
+  const firearmDay = cleanDay(firearmDate.getDate());
+  const firearmMonth = cleanMonth(firearmDate.getMonth() + 1);
+  const firearmYear = cleanYear(firearmDate.getFullYear());
+  const firearmDateInvalid =
+    firearmDate.getDate() !== firearmDay ||
+    firearmDate.getMonth() + 1 !== firearmMonth ||
+    firearmDate.getFullYear() !== firearmYear;
+
+  const qualificationDate = new Date(body.qualificationObtainedDate);
+  const qualificationDay = cleanDay(qualificationDate.getDate());
+  const qualificationMonth = cleanMonth(qualificationDate.getMonth() + 1);
+  const qualificationYear = cleanYear(qualificationDate.getFullYear());
+  const qualificationDateInvalid =
+    qualificationDate.getDate() !== qualificationDay ||
+    qualificationDate.getMonth() + 1 !== qualificationMonth ||
+    qualificationDate.getFullYear() !== qualificationYear;
+
+  return {
+    // The booleans are just copied across.
+    convictions: body.convictions,
+    bestPractice: body.bestPractice,
+    // The Integers need to be checked that they just contain numbers and are positive
+    certificateNumber: body.certificateNumber === undefined ? undefined : cleanInt(body.certificateNumber),
+    redExperience: body.redExperience === undefined ? undefined : cleanInt(body.redExperience),
+    redControl: body.redControl === undefined ? undefined : cleanInt(body.redControl),
+    roeExperience: body.roeExperience === undefined ? undefined : cleanInt(body.roeExperience),
+    roeControl: body.roeControl === undefined ? undefined : cleanInt(body.roeControl),
+    sikaExperience: body.sikaExperience === undefined ? undefined : cleanInt(body.sikaExperience),
+    sikaControl: body.sikaControl === undefined ? undefined : cleanInt(body.sikaControl),
+    fallowExperience: body.fallowExperience === undefined ? undefined : cleanInt(body.fallowExperience),
+    fallowControl: body.fallowControl === undefined ? undefined : cleanInt(body.fallowControl),
+    // The dates
+    certificateIssuedDate: firearmDateInvalid ? undefined : body.certificateIssuedDate,
+    qualificationObtainedDate: qualificationDateInvalid ? undefined : body.qualificationObtainedDate,
```

Comment:
Here you're returning the original strings, instead of the parsed Date objects.

Up here
https://github.com/Scottish-Natural-Heritage/fit-and-competent-api/pull/5/files#diff-dbc939d8bf57a7bd48e2ca8ca89fffdc3c47dca5f914998e731a208c16518391R21-R26
and here
https://github.com/Scottish-Natural-Heritage/fit-and-competent-api/pull/5/files#diff-dbc939d8bf57a7bd48e2ca8ca89fffdc3c47dca5f914998e731a208c16518391R39-R44
you've declared the DB Model/Schema/etc. as Date types, not string types.

Shouldn't we send them the validated date objects since we've done the hard work of validating them already?